ui/src/timespan/DoneTrackers.tsx

```diff
@@ -125,14 +126,16 @@ export const DoneTrackers: React.FC<DoneTrackersProps> = ({addTagsToTracker}) =>
 const DatedTimeSpans: React.FC<{
     name: string;
     setHeight: (cb: (height: Record<string, number>) => Record<string, number>) => void;
+    height: number;
     timeSpans: TimeSpanProps[];
-} & DoneTrackersProps> = ({name, timeSpans, addTagsToTracker, setHeight}) => {
+} & DoneTrackersProps> = ({name, timeSpans, addTagsToTracker, setHeight, height}) => {
     const ref = React.useRef<HTMLDivElement | null>();
     React.useEffect(() => {
-        if (ref.current) {
-            setHeight((old) => ({...old, [name]: ref.current!.getBoundingClientRect().height}));
+        const currentHeight = ref.current && ref.current.getBoundingClientRect().height;
+        if (currentHeight != null && currentHeight !== height) {
```

Comment:
I've adjusted this, to check outside of the setState. The react docs weren't clear it guaranteed that setState with the same value doesn't force a rerender.

Reply:
It does skip rerendering: https://react.dev/reference/react/useState#setstate-caveats


> If the new value you provide is identical to the current state, as determined by an [Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison, React will skip re-rendering the component and its children. This is an optimization. Although in some cases React may still need to call your component before skipping the children, it shouldn’t affect your code.

Reply:
> Although in some cases React may still need to call your component before skipping the children

Okay, then "reexecute" the compomennt instead of rerender, but yeah doesn't really matter as the new impl. behaves the same.